.github/workflows/build-&-publish-docker-image.yml

```diff
@@ -90,27 +92,25 @@ jobs:
 
     - name: Build & Push Sharder Docker Image
       run: |
-        docker build --build-arg GIT_COMMIT=$GITHUB_SHA -t $SHARDER_REGISTRY:$TAG -f docker.local/build.sharder/Dockerfile .
-        docker tag $SHARDER_REGISTRY:$TAG $SHARDER_REGISTRY:latest
-        docker tag $SHARDER_REGISTRY:$TAG $SHARDER_REGISTRY:staging
-        docker push $SHARDER_REGISTRY:$TAG
-      env:
-        TAG: ${{ steps.get_info.outputs.IMAGE_TAG }}
+        docker build --build-arg GIT_COMMIT=$GITHUB_SHA -t sharder:latest -f docker.local/build.sharder/Dockerfile .
+
+        docker tag sharder:latest ${SHARDER_REGISTRY}:$TAG
+        docker push ${SHARDER_REGISTRY}:$TAG
+
+        SHORT_SHA=$(echo $GITHUB_SHA | head -c 8)
+        docker tag sharder:latest ${SHARDER_REGISTRY}:$TAG-$SHORT_SHA
+        docker push ${SHARDER_REGISTRY}:$TAG-$SHORT_SHA  
+
 
     - name: Push Sharder Docker Image With latest Tag
+      if: ${{ github.event.inputs.latest_tag == 'yes' || github.ref == 'refs/heads/master' }}
```

Comment:
can we remove this latest_tag logic please? it adds unnecessary complication 
latest tag should always be pushed

Reply:
shader:latest and miner:latest are not used on our development. I think devops team need them, so just leave it with @anish-squareops

Reply:
we require the latest tag logic as its for master code.